### ISSUE TITLE
`winget` fails on Windows for ARM

### ISSUE THREAD

**Issue by aslilac** [microsoft/winget-cli#1993]
### Brief description of your issue

When running any `winget` command I get...

```
Could not open the resource file: 0x80073b1f : ResourceMap Not Found.
```

### Steps to reproduce

`winget search git`, for example (or `winget search $anything`)

### Expected behavior

To be able to run `winget` to install things.

### Actual behavior

Fails to run any command, with all updates from Windows Update and the Store app installed.

### Environment

```shell
Edition	Windows 11 Home
Version	21H2
Installed on	‎2/‎14/‎2022
OS build	22000.527
Experience	Windows Feature Experience Pack 1000.22000.527.0

Can't run `winget --info`, it fails with the same error as every other command.

Running in Parallels on an M1 Max if any of that might be relevant.
```

**Comment by jedieaston** [comment]
Does this get resolved for you if you install winget manually from [GitHub Releases](https://github.com/microsoft/winget-cli/releases/tag/v1.1.12653)? I ask because I could replicate the same behavior on my ARM system, but it seemed to be because I tried to run winget when the install from the Microsoft Store wasn't complete (the exe was present but some of the DLLs were not). After waiting a minute and typing winget again, it worked. 

I don't think your issue is the same, but I'm wondering if the package didn't install right.

**Comment by aslilac** [comment]
<img width="601" alt="Screen Shot 2022-03-04 at 9 17 04 AM" src="https://user-images.githubusercontent.com/418348/156799601-08a5cae6-88ca-4fb2-86ea-09c94c31034b.png">

Installing from Github releases doesn't seem to work 🤔

**Comment by denelon** [comment]
@jedieaston, can you test with 1.2 and confirm if the problem of dependencies is resolved? We may need to have a servicing release if it's not.

**Comment by denelon** [comment]
I need to get a Mac with Parallels.

**Comment by jedieaston** [comment]
> @jedieaston, can you test with 1.2 and confirm if the problem of dependencies is resolved? We may need to have a servicing release if it's not.

Package dependencies seem to work for me on 1.2 on ARM64:
<img width="1316" alt="image" src="https://user-images.githubusercontent.com/21368066/156819948-164bed61-bd18-4bf4-8aeb-f8014e54cc3e.png">

(Were you referring to winget dependencies or dependencies for App Installer?)

**Comment by denelon** [comment]
I was referring to Windows Package Manager dependencies in App Installer.

**Comment by jedieaston** [comment]
It seems like they are automatically resolved just like on other platforms (UI Xaml has to be around for Windows 11 UI stuff). I didn't have to manually install anything.

**Comment by TechPrototyper** [comment]
> I need to get a Mac with Parallels.

Unfortunately, this won't help. I am sitting right in front of one. And get the same error, trying to run latest version of API Management Resource Kit.

**Comment by atwright147** [comment]
I am having this problem too

**Comment by denelon** [comment]
Does the "Microsoft Store" work on Parallels?

I'm wondering if the dependency management isn't working correctly, or correctly detecting the architecture. We added the architecture as a display property for `winget --info` so we can show when we're running on ARM hardware natively.

**Comment by denelon** [comment]
I haven't tested on Parallels. I'll see if I can find someone here to test/validate.
I have tested WinGet on Windows natively running on ARM64.

**Comment by jedieaston** [comment]
Ugh, my Parallels license expired, but I can confirm winget also works installed from the Microsoft Store in a VM (using UTM, which uses the Apple Hypervisor just like Parallels):

<img width="1002" alt="image" src="https://user-images.githubusercontent.com/21368066/196983951-a11f1c5a-a8e6-426e-9ba2-9654ec1cf39c.png">

I'm on Build 25227 (Dev Insider Preview).
@atwright147, what build of Windows are you running on (`winver`)?

**Comment by atwright147** [comment]
Hey, not sure if I am out of date or not but `Version 21H2 (OS Build 22000.978)`

![Screenshot 2022-10-20 at 17 56 35](https://user-images.githubusercontent.com/986665/197012205-923f6ff2-9aeb-41f4-83fa-0032654a086d.png)

**Comment by jedieaston** [comment]
You're not on Insiders, but this should be working fine anyway. Will have to keep digging, unless @denelon finds something first.

**Comment by atwright147** [comment]
@jedieaston I am happy to do so but Windows 11 Installation Assistant doesn't seem to work on Windows 11 for ARM64

When I run it, I get the following message:

> Installation Assistant doesn't work on this type of device
>
> For other ways to install Windows 11 on this ARM64 device, visit [Download Windows 11](https://www.microsoft.com/en-us/windows/windows-11?r=1)

**Comment by jedieaston** [comment]
> @jedieaston I am happy to do so but Windows 11 Installation Assistant doesn't seem to work on Windows 11 for ARM64
> 
> When I run it, I get the following message:
> 
> > Installation Assistant doesn't work on this type of device
> > For other ways to install Windows 11 on this ARM64 device, visit [Download Windows 11](https://www.microsoft.com/en-us/windows/windows-11?r=1)
> 
> I guess I could join the Insider Program if that would help?

I didn't know that, sorry (I use Windows 11 on ARM64 occasionally, but usually when I need a new version I rebuild my VM). 

If you want to you can join Insiders, but you shouldn't have to. Is Windows Update offering you an update to 22H2?

**Comment by Masamune3210** [comment]
Could always try manually installing the msu, should be able to still grab it from the Update Catalog

**Comment by atwright147** [comment]
msu?

**Comment by Masamune3210** [comment]
Microsoft Update, its the packaged version of a update designed for either sysadmins that want to manually install a specific update or for people who are offline and need a way to install the updates without having access to the normal way of doing it.

https://www.catalog.update.microsoft.com/Home.aspx

**Comment by atwright147** [comment]
I went to https://www.catalog.update.microsoft.com/Search.aspx?q=22h2 and downloaded the top Win 11 for ARM64 22H2 file.

I only have the option to extract files from the CAB file and once extracted, the cannot install (`setup.exe`) due to not being able to determine the language. Super frustrating. ARM64 in Parallels seems a bit borked.

**Comment by Masamune3210** [comment]
Sounds like it unfortunately

**Comment by atwright147** [comment]
Right, so on the basis that something must have been fully broken, I upgraded to Parallels v18.1.0 (53311).

I deleted the previous Win11 VM, then created a new Win11 VM through Parallels automatic system and it all works flawlessly now :)

**Comment by Masamune3210** [comment]
Yeah, it sounds like that previous install was just completely hosed in some way that stopped winget from functioning correctly. Glad you got it working

**Comment by denelon** [comment]
@aslilac are you still having trouble running on ARM?

**Comment by aslilac** [comment]
it seems to be working now!
